ansible/roles/test/files/ptftests/arista.py

```diff
@@ -226,7 +231,10 @@ def parse_logs(self, data):
 
         result['route_timeout'] = result_rt
 
-        if initial_time_bgp == -1 or initial_time_if == -1:
+        # for fast-reboot, we expect to have both the bgp and portchannel events in the logs. for warm-reboot, portchannel events might not be present in the logs all the time.
+        if self.reboot_type == 'fast-reboot' and (initial_time_bgp == -1 or initial_time_if == -1):
+            return result
+        elif self.reboot_type == 'warm-reboot' and (initial_time_bgp == -1 and initial_time_if == -1):
```

Comment:
Why for WR it's an issue if both bgp and LAG shows no messages?
If just bgp shows nothing, or LAG shows nothing this is alright?

Reply:
For WR, bgp info will always show up in the logs because of the time diff between the idle and established messages. So I expect it to be there all the time. So the 'bgp shows nothing' should always be false. But portchannel messages are seen sometimes with same timestamp for up to down or a diff of 1s (Mellanox).

Reply:
Why do we need anything which is constant in the formula in this case?

Reply:
I didn't get your point.  Are you referring to the -1? That value is returned by the extract_from_logs() if regex match fails.
I couldn't figure out any other way to do it. Let me know if you have any suggestions.

Reply:
Fix me if I wrong:

From your words: 

> For WR, bgp info will always show up in the logs because of the time diff between the idle and established messages.

That means that `initial_time_bgp == -1` will never be True ?

Reply:
In the ideal case, it will never be true. if its true, then I don't have the bgp info in the logs. I need to return because I can't continue with any more bgp related checks.

Reply:
As discussed, removed the portchannel check for warm-reboot case